### ISSUE TITLE
Plot warmup samples in traceplot

### ISSUE THREAD

**Issue by rachelstephlee** [arviz-devs/arviz#1621]
I'm working with a fit from Pystan. When converting to InferenceData, I can use the flag 'save_warmup', but even if I do that, I don't understand what the significance of that would do. 

I'm trying to get a traceplot that is similar to how R would plot it-- giving me a choice to visualize the warmup samples as well. Am I missing something?

**Comment by rachelstephlee** [comment]
OK, I figured out that given some Inference data 'data', I can get the warmup posterior by calling 'data.warmup_posterior'. But is there a way to plot the warmup AND the regular samples together?

**Comment by ahartikainen** [comment]
Hi, there is no "build in" way to do that, but I think you can achieve similar plot with couple of different ways

Manually combine values with `az.concat` -> not the best solution, no difference between warmup and non-warmup.

Change coordinate values for `draws` as (-nwarmup + index) and plot traceplot for both groups using same axis.

cc. @OriolAbril or @percygautam do you have time to test this? (Are we using coordinate values for traceplot?)

**Comment by rachelstephlee** [comment]
Thank you Ari. Yes, I think i have hacked together something that looks like that using `az.concat`. Can you let me know about the coordinate values? If I can have that figured out, I should be all set.

**Comment by OriolAbril** [comment]
Yes, `plot_trace` uses the values of the draw coordinate as x values. I'd recommend the 2nd option Ari suggested, you can also use inferencedata level functions to update all warmup groups at the same time, the code could look like:

```
idata = idata.map(lambda ds: ds.assign_coords({"draw": ds.draw - len(ds.draw)}), groups="warmup", filter_groups="like")
ax = az.plot_trace(idata)
ax = az.plot_trace(idata, group="warmup_posterior", ax=ax)
```

hopefully the code won't have some hardcoded xlims (which we should probably remove if they are), if there are you'd have to loop over the 2nd colum of `ax` to set the lims to the right values as xlim
